internal/service/projectapikey/resource_project_api_key.go

```diff
@@ -197,9 +197,12 @@ func resourceMongoDBAtlasProjectAPIKeyUpdate(ctx context.Context, d *schema.Reso
 			for _, apiKey := range newAssignments {
 				projectID := apiKey.(map[string]any)["project_id"].(string)
 				roles := conversion.ExpandStringList(apiKey.(map[string]any)["role_names"].(*schema.Set).List())
-				_, err := conn.ProjectAPIKeys.Assign(ctx, projectID, apiKeyID, &matlas.AssignAPIKey{
-					Roles: roles,
-				})
+				assignment := []admin.UserAccessRoleAssignment{
+					{
+						Roles: &roles,
+					},
+				}
+				_, _, err := connV2.ProgrammaticAPIKeysApi.AddProjectApiKey(ctx, projectID, apiKeyID, &assignment).Execute()
```

Comment:
Found it a bit verbose, maybe one line is better?
```suggestion
				assignment := []admin.UserAccessRoleAssignment{{Roles: &roles}}
```

Reply:
Done in all the cases, thanks for the suggestion